src/mlpack/methods/bayesian_linear_regression/bayesian_linear_regression.hpp

```diff
@@ -0,0 +1,285 @@
+ * The determination of \f$ alpha \f$ and of the noise precision \f$ beta \f$
+ * is part of the optimization process, leading to an automatic determination of 
+ * w. The model being entirely based on probabilty distributions, uncertainties 
+ * are available and easly computed for both the parameters and the predictions.
+ *
+ * The advantage over linear regression and ridge regression is that the 
+ * regularization is determined from all the training data alone without any
+ * require to an hold out method. 
+ *
+ * The code below is an implementation of the maximization of the evidence 
+ * function described in the section 3.5.2 of the C.Bishop book, Pattern
+ * Recognition and Machine Learning.  
+ *
+ * @code
+ * @article{MacKay91bayesianinterpolation,
+ *   author = {David J.C. MacKay},
+ *   title = {Bayesian Interpolation},
+ *   journal = {NEURAL COMPUTATION},
+ *   year = {1991},
+ *   volume = {4},
+ *   pages = {415--447}
+ * }
+ * @endcode
+ *
+ * @code
+ * @book{Bishop:2006:PRM:1162264,
+ *   author = {Bishop, Christopher M.},
+ *   title = {Pattern Recognition and Machine Learning (Information Science 
+ *            and Statistics)},
+ *   chapter = {3}
+ *   year = {2006},
+ *   isbn = {0387310738},
+ *   publisher = {Springer-Verlag},
+ *   address = {Berlin, Heidelberg},
+ * } 
+ * @endcode
+ *   
+ * Example of use:
+ *
+ * @code
+ * arma::mat xTrain; // Train data matrix. Column-major.
+ * arma::rowvec yTrain; // Train target values.
+ 
+ * // Train the model. Regularization strength is optimally tunned with the
+ * // training data alone by applying the Train method.
+ * BayesianLinearRegression estimator(); // Instanciate the estimator with default option.
+ * estimator.Train(xTrain, yTrain);
+ 
+ * // Prediction on test points.
+ * arma::mat xTest; // Test data matrix. Column-major.
+ * arma::rowvec predictions;
+ 
+ * estimator.Predict(xTest, prediction);
+ 
+ * arma::rowvec yTest; // Test target values.
+ * estimator.RMSE(xTest, yTest); // Evaluate using the RMSE score.
+ 
+ * // Compute the standard deviations of the predictions.
+ * arma::rowvec stds;
+ * estimator.Predict(xTest, responses, stds)
+ * @endcode
+ */
+class BayesianLinearRegression
+{
+ public:
+  /**
+   * Set the parameters of Bayesian Ridge regression object. The
+   * regularization parameter is automatically set to its optimal value by
+   * maximization of the marginal likelihood.
+   *
+   * @param centerData Whether or not center the data according to the
+   *    examples.
+   * @param scaleData Whether or not scale the data according to the
+   *    standard deviation of each feature.
+   * @param nIterMax Maximum number of iterations for convergency.
+   * @param tol Level from which the solution is considered sufficientlly 
+   *    stable.  
+   */
+  BayesianLinearRegression(const bool centerData = true,
+                           const bool scaleData = false,
+                           const int nIterMax = 50,
```

Comment:
Do you mind to go with `size_t` instead of `int` here, just to be consistent with the rest of the codebase.

Reply:
Yes of course. I tried to test the code once modified before to push but I faced some compilation errors. Feel free to modify it directly if you want :+1: